list.js

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* eslint max-statements:[1, 30] */
+'use strict';
+
+var bufrw = require('bufrw');
+var TYPE = require('./TYPE');
+var typeNames = require('./names');
+var InvalidSizeError = require('./errors').InvalidSizeError;
+var ListTypeIdMismatch = require('./errors').ListTypeIdMismatch;
+
+var LengthResult = bufrw.LengthResult;
+var WriteResult = bufrw.WriteResult;
+var ReadResult = bufrw.ReadResult;
+
+function ListSpec(valueType, annotations) {
```

Comment:
Not sure what you’re proposing for stash. Upcoming, SetSpec will be the first illustration of how they are used.